### PR TITLE
feat(theme-default): improve SidebarButton a11y and animate

### PR DESCRIPTION
**Pull request by jiazengp**
Animation effect:

<video src="https://user-images.githubusercontent.com/86162890/128382982-1ee82108-42a7-4083-a6b8-d0cde3d5ab3c.mp4"></video>

### CODE REVIEW

File: packages/@vuepress/theme-default/src/shared/options.ts [hunk 1]
```diff
@@ -253,6 +253,11 @@ export interface DefaultThemeLocaleData extends LocaleData {
    */
   openInNewWindow?: string
 
+  /**
+   * Toggle sidebar button a11y text
+   */
+  sidebarButtonText?: string
```
Comment (meteorlxy) [inline packages/@vuepress/theme-default/src/shared/options.ts:259]: Should use `toggleSidebar` to keep align with `toggleDarkMode`, and so does the comment

File: packages/@vuepress/theme-default/src/client/components/ToggleSidebarButton.vue [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
+  setup() {
+    const title = computed(() => {
+      const themeLocale = useThemeLocaleData()
+      return themeLocale.value.sidebarButtonText || 'Toggle Sidebar'
+    })
+
+    return {
+      title,
+    }
+  },
```
Comment (meteorlxy) [inline packages/@vuepress/theme-default/src/client/components/ToggleSidebarButton.vue:36]: Let's migrate to setup script

**Review by meteorlxy — CHANGES_REQUESTED** [review]

File: packages/@vuepress/theme-default/src/client/components/ToggleSidebarButton.vue (hunk 2 above)
Comment (jiazengp) [reply packages/@vuepress/theme-default/src/client/components/ToggleSidebarButton.vue]: All right👌

**Comment by jiazengp** [comment]
😀I also added the document @meteorlxy

**Comment by meteorlxy** [comment]
Sorry for delay. Busy these days. I'll review this PR later

**Comment by jiazengp** [comment]
It’s okay, I’m just worried you forgot it😁

File: packages/@vuepress/theme-default/src/shared/options.ts (hunk 1 above)
Comment (jiazengp) [reply packages/@vuepress/theme-default/src/shared/options.ts]: All right

**Comment by meteorlxy** [comment]
????

What happened to this PR?

I did nothing but it is marked as merged?

@jiazengp I think that the pull bot broke your PR:

image

**Comment by meteorlxy** [comment]
The pull bot made this PR empty:

image

**Comment by jiazengp** [comment]
啊，这

**Comment by jiazengp** [comment]
抱歉，我重新提一个~。

**Pull request merged**